src/contractnetwork.jl

```diff
@@ -0,0 +1,47 @@
+using BackendSelection: @Algorithm_str, Algorithm
+using ITensorNetworksNext.LazyNamedDimsArrays: substitute, materialize, lazy,
+    symnameddims
+
+#Algorithmic defaults
+default_sequence_alg(::Algorithm"exact") = "leftassociative"
+default_sequence(::Algorithm"exact") = nothing
+function set_default_kwargs(alg::Algorithm"exact")
+    sequence = get(alg, :sequence, nothing)
+    sequence_alg = get(alg, :sequence_alg, default_sequence_alg(alg))
+    return Algorithm("exact"; sequence, sequence_alg)
+end
+
+function contraction_sequence_to_expr(seq)
+    if seq isa AbstractVector
+        return prod(contraction_sequence_to_expr, seq)
+    else
+        return symnameddims(seq)
+    end
+end
+
+function contraction_sequence(::Algorithm"leftassociative", tn::Vector{<:AbstractArray})
+    return prod(symnameddims, 1:length(tn))
+end
+
+function contraction_sequence(tn::Vector{<:AbstractArray}; sequence_alg = default_sequence_alg)
```

Comment:
Should this be:
```suggestion
function contraction_sequence(tn::Vector{<:AbstractArray}; sequence_alg = default_sequence_alg(Algorithm("exact"))
```
or:
```suggestion
function contraction_sequence(tn::Vector{<:AbstractArray}; sequence_alg = "leftassociative")
```
?

Reply:
I guess either would work, but I went with the former.